lib/needle.js

```diff
@@ -544,14 +545,18 @@ Needle.prototype.send_request = function(count, method, uri, config, post_data,
           debug('Clearing original request cookie', config.headers['cookie']);
           delete config.headers['cookie'];
         }
-
         if (config.follow_set_referer)
           config.headers['referer'] = encodeURI(uri); // the original, not the destination URL.
 
         config.headers['host'] = null; // clear previous Host header to avoid conflicts.
 
         debug('Redirecting to ' + url.resolve(uri, headers.location));
+        if(!config.follow_authorization_header){
+            delete config.headers['authorization']
+            delete config.headers['Authorization']
+        }
         return self.send_request(++count, method, url.resolve(uri, headers.location), config, post_data, out, callback);
+    ;
```

Comment:
what is this ; and what does it do?

Reply:
@Kalininator its a terrible mistake .......